### ISSUE TITLE
Support `module:entry_point` in addition to `script.py` on command line

### ISSUE THREAD

**Issue by callegar** [pyinstaller/pyinstaller#8074]
**Is your feature request related to a problem? Please describe.**

I would like to be able to use pyinstaller on a module that has an entry point without the need to create a wrapper script.

**Describe the solution you'd like**
I would like to be able to say `pyinstaller -F mymodule:main -n mytool` rather than `pyinstaller mytool.py` which implies the need to create `mytool.py`.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by bwoodsend** [comment]
Doesn't `mymodule` have a `__main__.py` submodule which runs the entry-point function? If it does, just run PyInstaller on that. If it doesn't then add it so that people who don't have their scripts/bin directory in `PATH` are able to run the entry-point (via `python -m mymodule`).

**Comment by callegar** [comment]
It doesn't, because the package has multiple entry points. Furthermore, I do not like very much the idea of adding `__main__.py`, because I would not like to touch the package and because the package might need to get a `__main__.py` in the future, exposing a "default" entry point different from the one I need now.

The fact is that `setuptool`, the `pyproject` standardization, `poetry`, `setuptools`, etc., all now tend to support the idea of packages with (multiple) entry points declarable via `pyproject.toml` (e.g., via a `[tool.poetry.scripts]` section in poetry, or a more standard `[project.scripts]` entry otherwise). In all cases the syntax is `module:callable` (or `package.module:callable`, if the module is inside a package). Would be great to have this syntax supported in `pyinstaller` too, for homogeneity, and for being able to use pyinstaller on packages one would prefer not to modify.

**Comment by rokm** [comment]
On one hand, this seems like a non-trivial change to save the user from creating a two-line entry-point script for their frozen application. 

On the other hand, having this would allow us to get rid of that horrible traversal of parent directories when entry-point .py file is in a package (has `__init__.py` next to it), and just raise an error instead.

So while this is not getting an outright rejection from me, it is, as far as I am concerned, quite low on the priority list.

**Comment by gtoombs-avidrone** [comment]
> Doesn't `mymodule` have a `__main__.py` submodule which runs the entry-point function? If it does, just run PyInstaller on that.

That's not enough. `__main__.py` may have relative `.` imports under the assumption that it actually lives in a module, and if I'm not mistaken, directly passing the path to `__main__.py` loads it as a script and not a module, which is guaranteed to break several assumptions about Python module structure; this will only work by luck. First-class support for module-level entry point would be very valuable.

**Comment by C0rn3j** [comment]
Related discussion in https://github.com/orgs/pyinstaller/discussions/9103

TL;DR this setup worked for me while still keeping Tauon as a module:

`pyproject.toml`:

```toml
[project.gui-scripts]
	tauonmb = "tauon.__main__:main"
```

`__main__.py`:
```python
if __name__ == "__main__" and not __package__:
	__package__ = "tauon"
```

```python
#from tauon.t_modules import t_bootstrap
from .t_modules import t_bootstrap
````

`linux.spec`:

```python
#["src/tauon/__main__.py"],
[".venv/bin/tauonmb"],
```
